x-pack/plugins/security_solution/public/overview/components/overview_cti_links/cti_no_events.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { useCtiDashboardLinks } from '../../containers/overview_cti_links';
+import { ThreatIntelPanelView } from './threat_intel_panel_view';
+import { CtiInnerPanel } from './cti_inner_panel';
+import * as i18n from './translations';
+import { emptyEventCountsByDataset } from '../../containers/overview_cti_links/helpers';
+
+const warning = (
+  <CtiInnerPanel
+    color={'warning'}
+    title={i18n.WARNING_TITLE}
+    body={i18n.WARNING_BODY}
+    data-test-subj="cti-inner-panel-warning"
+  />
+);
+
+export const CtiNoEventsComponent = ({ to, from }: { to: string; from: string }) => {
+  const { buttonHref, listItems, isDashboardPluginDisabled } = useCtiDashboardLinks(
+    { ...emptyEventCountsByDataset },
+    to,
+    from
+  );
+
+  return (
+    <ThreatIntelPanelView
+      buttonHref={buttonHref}
+      listItems={listItems}
+      splitPanel={warning}
+      totalEventCount={0}
+      isDashboardPluginDisabled={isDashboardPluginDisabled}
+    />
+  );
+};
+
+CtiNoEventsComponent.displayName = 'CtiNoEvents';
+
+export const CtiNoEvents = React.memo(CtiNoEventsComponent);
```

Comment:
How are you deciding when to use React.memo for these components? Asking for a friend 😉

Reply:
[this](https://dmitripavlutin.com/use-react-memo-wisely/) unofficial guide seems to have some good information - I see React.memo being used in the codebase liberally, and I haven't hesitated to add it to a component with no props as that seems to be very safe